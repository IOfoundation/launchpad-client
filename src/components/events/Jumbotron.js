import React from 'react';

const Jumbotron = () => {
  return (
    <div className="events-jumbotron">
      <div className="events-jumbotron__container">
        <h2 className="events-jumbotron__title">{"What's Happening"}</h2>
        <p className="events-jumbotron__description">
          {
            'There are many events throughout the Sacramento Region for all phases of businesses. See our most popular keynote events on the event list below or explore the calendar.'
          }
        </p>
      </div>
    </div>
  );
};

export default Jumbotron;
