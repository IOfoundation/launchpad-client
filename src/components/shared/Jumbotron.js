import React from 'react';
import {PropTypes} from 'prop-types';

const Jumbotron = props => {
  const {descriptionModifacor, title, description, background} = props;
  const descriptionClasses = ['jumbotron__description', 'text-thin'];

  if (props.descriptionModifacor) {
    descriptionClasses.push(`jumbotron__description--${descriptionModifacor}`);
  }

  return (
    <div
      className="jumbotron"
      style={{
        backgroundImage: `url(/static-data/images/${background}.png)`,
      }}
    >
      <div className="jumbotron__container">
        <h2 className="jumbotron__title">{title}</h2>
        <p className={descriptionClasses.join(' ')}>{description}</p>
      </div>
    </div>
  );
};

Jumbotron.propTypes = {
  background: PropTypes.string,
  description: PropTypes.string,
  descriptionModifacor: PropTypes.string,
  title: PropTypes.string,
};

export default Jumbotron;
