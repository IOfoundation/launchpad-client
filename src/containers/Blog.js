import React, {PureComponent} from 'react';
import MainLayout from '../components/layouts/Main';
import BlogLayout from '../components/blog/Blog';
import {viewport, sizeCheck} from '../utils';

class Blog extends PureComponent {
  state = {
    width: viewport().width,
    breakpoint: '',
    homePage: false,
    listener: () => sizeCheck(this.handleWindowSizeChange),
  };

  componentDidMount() {
    window.addEventListener('resize', this.state.listener);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.state.listener);
  }

  handleWindowSizeChange = breakpoint => {
    this.setState({breakpoint, width: viewport().width});
  };

  render() {
    const {width, homePage, breakpoint} = this.state;

    return (
      <MainLayout windowWidth={width} homePage={homePage}>
        <BlogLayout breakpoint={breakpoint} />
      </MainLayout>
    );
  }
}

export default Blog;
