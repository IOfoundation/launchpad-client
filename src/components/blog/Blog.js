import React from 'react';

import Jumbotron from '../shared/Jumbotron';
import FeaturedPosts from './FeaturedPosts';

const Blog = () => {
  return (
    <div className="blog-container">
      <Jumbotron
        title="Resource Finder Blog"
        description="Find content relevant to your industry and business stage."
        descriptionModifacor="full-white"
      />
      <FeaturedPosts />
    </div>
  );
};

export default Blog;
