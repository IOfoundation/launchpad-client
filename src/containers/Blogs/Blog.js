import React from 'react';
import {PropTypes} from 'prop-types';
import Layout from '../Layout';
import Blog from '../../components/blog/Blog';

const BlogRoute = props => {
  return (
    <Layout>
      <Blog breakpoint={props.breakpoint} />
    </Layout>
  );
};

BlogRoute.propTypes = {
  breakpoint: PropTypes.string,
};

export default BlogRoute;
