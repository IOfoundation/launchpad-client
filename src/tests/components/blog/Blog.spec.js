import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import Blog from '../../../components/blog/Blog';
import Jumbotron from '../../../components/shared/Jumbotron';

const props = {
  breakpoint: 'sm',
};

describe('<Blog />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Blog {...props} />);
  });

  it('renders snapshot of Blog', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should create Jumbotron with props', () => {
    const jumbotron = wrapper.find(Jumbotron);

    expect(jumbotron.props()).toEqual({
      title: 'Resource Finder Blog',
      description: 'Find content relevant to your industry and business stage.',
      descriptionModifacor: 'full-white',
      background: 'blog-header',
    });
  });
});
