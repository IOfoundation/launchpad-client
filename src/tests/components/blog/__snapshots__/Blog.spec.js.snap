// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blog /> renders snapshot of Blog 1`] = `
<div
  className="blog-container"
>
  <Jumbotron
    background="blog-header"
    description="Find content relevant to your industry and business stage."
    descriptionModifacor="full-white"
    title="Resource Finder Blog"
  />
  <WithStyles(Connect(withRouter(FeaturedPosts))) />
  <WithStyles(MainContent)
    breakpoint="sm"
  />
</div>
`;
